code/game/objects/structures/window.dm

```diff
@@ -336,13 +338,18 @@
 	update_nearby_icons()
 
 /obj/structure/window/proc/spawnDebris(location)
+	var/datum/material/glass_material_ref = GET_MATERIAL_REF(glass_material_datum)
+	var/obj/item/shard_type = glass_material_ref.shard_type
+	var/obj/effect/decal/debris_type = glass_material_ref.debris_type
 	. = list()
-	. += new /obj/item/shard(location)
-	. += new /obj/effect/decal/cleanable/glass(location)
+	if(!isnull(shard_type))
+		. += new shard_type(location)
+		if (fulltile)
+			. += new shard_type(location)
+	if(!isnull(debris_type))
+		. += new debris_type(location)
 	if (reinf)
 		. += new /obj/item/stack/rods(location, (fulltile ? 2 : 1))
```

Comment:
the more i look at this the less i like the dot-variable meme (we don't need it anymore since we don't really override this any more), just make it like

```txt
var/list/returnable_list = list()
// add stuff to returnable_list and all that cool shit
return returnable_list
```

Much more explicit and kinder to those who don't understand dot memes, especially since we aren't doing something like Examine() where we have to invoke parent via `. = ..()` and then add onto the dot variable meme however we want- it's all self-contained in this proc now.

Optionally if you want to go the distance, it would be nice if you could also DMDoc spawnDebris (and brownie points for making it snake_case with spawn_debris)